### PR TITLE
Fixes when referring to entity YAML filename to use new standard catalog-info.yaml

### PR DESCRIPTION
**Pull request by adamdmharvey**
## Hey, I just made a Pull Request!

* Fixes a few legacy references in doco of `component-info.yaml` to use the new `catalog-info.yaml`
* Patch to the `scaffolder-backend` so React templates get the proper entity YAML file name
* Fixed a minor typo in CircleCI doco which skips through the Vale checks

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: docs/features/software-catalog/descriptor-format.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ software catalog API.
   "kind": "Component",
   "metadata": {
     "annotations": {
-      "backstage.io/managed-by-location": "file:/tmp/component-info.yaml",
+      "backstage.io/managed-by-location": "file:/tmp/catalog-info.yaml",
```
Comment (adamdmharvey) [inline docs/features/software-catalog/descriptor-format.md:46]: I haven't tried this example, but I'll be honest I would have expected this to be in the format `file://` (double slash)?

Comment (Fox32) [reply docs/features/software-catalog/descriptor-format.md:46]: No it's not a `file://` url schema but a backstage type `file:` location. However I'm not sure if they are still supported or if they are now part of the type `url:`

**Comment by freben** [comment]
Hi! This has been fixed as part of https://github.com/backstage/backstage/pull/3388

**Comment by freben** [comment]
Hm buut there may be more added value on top of that one too, ping @benjdlambert

**Review by benjdlambert — APPROVED** [review]
I can merge both. The document updates sound better.

**Comment by adamdmharvey** [comment]
cheers - thanks gents!

**Pull request merged**